rl4co/data/transforms.py

```diff
@@ -83,54 +87,60 @@ def symmetric_augmentation(xy: Tensor, num_augment: int = 8, first_augment: bool
     return symmetric_transform(x, y, phi[:, None, None])
 
 
-def env_aug_feats(env_name: str = None):
-    """What features to augment for a given environment
-    Usually, locs already includes depot, so we don't need to augment depot
-    """
-    return ["locs"]
-
-
 def min_max_normalize(x):
     return (x - x.min()) / (x.max() - x.min())
 
 
+def get_augment_function(augment_fn: Union[str, callable]):
+    if callable(augment_fn):
+        return augment_fn
+    if augment_fn == "dihedral8":
+        return dihedral_8_augmentation_wrapper
+    if augment_fn == "symmetric":
+        return symmetric_augmentation
+    raise ValueError(f"Unknown augment_fn: {augment_fn}")
+
+
 class StateAugmentation(object):
     """Augment state by N times via symmetric rotation/reflection transform
 
     Args:
-        env_name: environment name
         num_augment: number of augmentations
-        use_dihedral_8: whether to use dihedral_8_augmentation.  If True, then num_augment must be 8
+        augment_fn: augmentation function to use, e.g. 'symmetric' (default) or 'dihedral8', if callable, 
+            then use the function directly. If 'dihedral8', then num_augment must be 8
         normalize: whether to normalize the augmented data
         feats: list of features to augment
     """
 
     def __init__(
         self,
-        env_name: str = None,
         num_augment: int = 8,
-        use_dihedral_8: bool = False,
+        augment_fn: Union[str, callable] = 'symmetric', 
+        first_aug_identity: bool = True,
         normalize: bool = False,
         feats: list = None,
     ):
+        self.augmentation = get_augment_function(augment_fn)
         assert not (
-            use_dihedral_8 and num_augment != 8
-        ), "If `use_dihedral_8` is True, then num_augment must be 8"
-        if use_dihedral_8:
-            self.augmentation = dihedral_8_augmentation_wrapper
-        else:
-            self.augmentation = symmetric_augmentation
-
-        self.feats = env_aug_feats(env_name) if feats is None else feats
+            self.augmentation == dihedral_8_augmentation_wrapper and num_augment != 8
+        ), "If using the `dihedral8` augmentation function, then num_augment must be 8"
+
+        if feats is None:
+            log.info("Default augment feature: `locs`")
+        self.feats = ["locs"] if feats is None else feats
         self.num_augment = num_augment
         self.normalize = normalize
+        self.first_aug_identity = first_aug_identity
 
     def __call__(self, td: TensorDict) -> TensorDict:
         td_aug = batchify(td, self.num_augment)
         for feat in self.feats:
+            init_aug_feat = td_aug[feat][list(td.size()), 0].clone()
```

Comment:
[minor] We can make it more efficient by:
1. Moving the cloning part and setting outside of the loop
2. Do not call clone if `first_aug_identity` is set to `False`